pkg/metrics/metrics.go

```diff
@@ -0,0 +1,52 @@
+package metrics
+
+import (
+	"fmt"
+	logger "github.com/intel/cri-resource-manager/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	builtInCollectors map[string]InitCollector
+	log               = logger.NewLogger("metrics")
+)
+
+func init() {
+	builtInCollectors = make(map[string]InitCollector)
+}
+
+type InitCollector func() (prometheus.Collector, error)
+
+func RegisterCollector(name string, init InitCollector) error {
+	log.Info("registering collector %s...", name)
+
+	if _, fn := builtInCollectors[name]; fn {
+		return metricsError("Collector %s already registered", name)
+	}
+
+	builtInCollectors[name] = init
+
+	return nil
+}
+
+var registeredCollectors = []prometheus.Collector{}
```

Comment:
nit: I'd rather have one `var` section per module to observe what constitutes module's state in one place.

Reply:
+1